build-logic/src/main/kotlin/CompilerOptions.kt

```diff
@@ -39,8 +48,11 @@ fun KotlinCommonCompilerOptions.configure(target: Int, kotlinCompilerOptions: Ko
   when (this) {
     is KotlinJvmCompilerOptions -> {
       freeCompilerArgs.add("-Xjvm-default=all")
-      freeCompilerArgs.add("-Xjdk-release=${target.toJvmTarget().target}")
-      jvmTarget.set(target.toJvmTarget())
+      if (!isAndroid) {
+        // See https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/JavaCompileUtils.kt;l=410?q=Using%20%27--release%27%20option%20for%20JavaCompile%20is%20not%20supported%20because%20it%20prevents%20the%20Android%20Gradle%20plugin
+        freeCompilerArgs.add("-Xjdk-release=${actualTarget.toJvmTarget().target}")
```

Comment:
Not 100% sure why this started being a problem with 8.8. AFAIK, this has always been a problem.